### PR TITLE
feat: draft beautiful dnd

### PR DESCRIPTION
**Pull request by lee-chase**
Contributes to #3478 

Using beautiful dnd.

Probably chopped out more than needed to make sense of this, need to understand if it is working and if we've lost any features.

#### What did you change?

Datagrid customisable columns

#### How did you test and verify your work?

Storybook

### CODE REVIEW

File: packages/ibm-products/src/components/Datagrid/Datagrid/DraggableElement.js [hunk 1]
```diff
@@ -6,89 +6,24 @@
  * US Government Users Restricted Rights - Use, duplication or disclosure
  * restricted by GSA ADP Schedule Contract with IBM Corp.
  */
-import * as React from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import { Draggable, Locked } from '@carbon/react/icons';
-// import { useDrag, useDrop } from 'react-dnd';
+import { Draggable as DraggableIcon, Locked } from '@carbon/react/icons';
```
Comment (lee-chase) [inline packages/ibm-products/src/components/Datagrid/Datagrid/DraggableElement.js:11]: Renamed as `Draggable` is component from `react-beautiful-dnd`

File: packages/ibm-products/src/components/Datagrid/Datagrid/DraggableElement.js [hunk 2]
```diff
@@ -6,89 +6,24 @@
...
 const DraggableElement = ({
-  // id,
+  id,
   index,
-  listData,
   children,
-  // type,
   disabled,
   ariaLabel,
-  onGrab,
```
Comment (lee-chase) [inline packages/ibm-products/src/components/Datagrid/Datagrid/DraggableElement.js:29]: Keyboard support direct from dnd library

File: packages/ibm-products/src/components/Datagrid/Datagrid/DraggableElement.js [hunk 3]
```diff
@@ -6,89 +6,24 @@
...
-  onArrowKeyDown,
-  isFocused,
   isSticky,
-  // moveElement,
   selected,
-  positionLabel = 'Current position {index} of {total}',
```
Comment (lee-chase) [inline packages/ibm-products/src/components/Datagrid/Datagrid/DraggableElement.js:35]: only used by keyboard support

File: packages/ibm-products/src/components/Datagrid/Datagrid/DraggableElement.js [hunk 4]
```diff
@@ -99,96 +34,56 @@ const DraggableElement = ({
...
-        // handle when focus move to li element
-        setIsFocusedOnItem(e.currentTarget === e.target);
-      }}
-    >
-      <span className={`${blockClass}__shared-ui--assistive-text`}>
-        {ariaLabel}
-      </span>
-      {isDragging && !isOver ? (
-        <div
-          ref={preview}
```
Comment (lee-chase) [inline packages/ibm-products/src/components/Datagrid/Datagrid/DraggableElement.js:152]: Preview not implemented in beautiful, needs a little thought as to how to implement. 

Also worth noting that dragged items are not visible outside of the modal.

File: packages/ibm-products/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js [hunk 5]
```diff
@@ -38,38 +27,47 @@ const Columns = ({
...
+    <div
+      className={`${blockClass}__customize-columns-column-list`}
+      style={{ '--scroll-top': `${scrollTopCSS}px` }}
+      ref={listRef}
+    >
       <ol
         className={`${blockClass}__customize-columns-column-list--focus`}
         role="listbox"
         aria-describedby={`${blockClass}__customize-columns--instructions`}
-        onKeyDown={(e) => {
```
Comment (lee-chase) [inline packages/ibm-products/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:65]: keyboard provided by react-beautiful-dnd

**Comment by elycheea** [comment]
Closing this in favor of #3513

**Pull request closed without merging**